java/client/src/org/openqa/selenium/AppdynamicsCapabilities.java

```diff
@@ -0,0 +1,97 @@
+package org.openqa.selenium;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Map;
+
+/**
+ * Created by shashwat.srivastava on 5/30/17.
+ */
+public class AppdynamicsCapabilities {
+  /**
+   * AppDynamics extensions
+   *
+   * This capability forwards all commands to a specified end point before executing them.
+   *
+   * This capability affects the remote webdriver server, not the clients.
+   *
+   * @author Renault John Lecoultre <renault.lecoultre@appdynamics.com>
+   * @since 21/10/16
+   * Copyright 2016 AppDynamics inc. All rights reserved
+   */
+  public static  String COMMAND_WEBHOOK = "commandWebhook";
+  /**
+   * This capability sets the K9 output directory.
+   */
+  public static String K9_OUTPUT_DIR = "outputDir";
+  /**
+   * This capability sets the K9 test id
+   */
+  public static String K9_TEST_ID = "testId";
+
+  private String outputDir;
+  private String testID;
+  private String commandWebhook;
+  /**
+   * AppDynamics extension
+   *
+   * This capability will make it possible to set environment variables for
+   * the K9 process
+   */
+  public static final String APPDYNAMICS_CAPABILITIES = "appdynamicsCapabilities";
+
+  public AppdynamicsCapabilities(Map<String, String> map) {
+    if (map.containsKey(K9_OUTPUT_DIR) && map.get(K9_OUTPUT_DIR) != null) {
```

Comment:
Should we test for empty string? I don't remember if Selenium as Strings.isNullOrEmpty(), if it has it, let's use it.